openmmtools/mcmc.py

```diff
@@ -146,9 +146,19 @@ class MCMCMove(SubhookedABCMeta):
     statistics such as number of attempted moves and acceptance rates.
 
     """
+    def __init__(self, context_cache=None):
+        if context_cache is not None:
+            logger.warning("Ignoring context_cache argument. Specifying context_cache on" 
+                           "initialization of an MCMCMove is deprecated; please pass to"
+                           " apply() method as kwarg instead.")
+        # hardcoding unused attribute to global context cache. For compatibility.
+        self.context_cache = cache.global_context_cache
+        # private attribute for propagation context cache - useful for testing
+        self._propagation_context_cache = None
```

Comment:
What is this used for? Do we need it?

Reply:
So, we needed this for the test [here](https://github.com/choderalab/openmmtools/blob/clean-context-cache-usage/openmmtools/tests/test_mcmc.py#L255-L265). I'm refactoring this test to use the number of accesses instead of having to use this private attribute that's only useful in testing. In the end this attribute won't be needed, thanks!

Reply:
Now we don't need this attribute and testing is now using number of accesses.

Reply:
Shouldn't we delete the attribute then?